src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -44,10 +51,27 @@ public RenameFlyoutViewModel(InlineRenameSession session, TextSpan selectionSpan
             _session.ReferenceLocationsChanged += OnReferenceLocationsChanged;
             StartingSelection = selectionSpan;
             InitialTrackingSpan = session.TriggerSpan.CreateTrackingSpan(SpanTrackingMode.EdgeInclusive);
+            var smartRenameSession = smartRenameSessionFactory.Value.CreateSmartRenameSession(_session.TriggerSpan);
+            if (smartRenameSession is not null)
+            {
+                SmartRenameViewModel = new SmartRenameViewModel(smartRenameSession);
+                SmartRenameViewModel.PropertyChanged += OnSuggestedNameSelected;
+            }
 
             RegisterOleComponent();
         }
 
+        private void OnSuggestedNameSelected(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(SmartRenameViewModel.SelectedSuggestedName)
```

Comment:
i'm a bit confused about the need to check this property name.  what other events do we have?

Reply:
Will add comment about the events here